app/models/report.rb

```diff
@@ -478,8 +613,217 @@ def self.report_trending_search(report)
       report.data << {
         term: trend.term,
         unique_searches: trend.unique_searches,
-        ctr: (ctr * 100).ceil(1).to_s + "%"
+        ctr: (ctr * 100).ceil(1)
+      }
+    end
+  end
+
+  def self.report_moderators_activity(report)
+    report.labels = [
+      { type: :link, properties: [:username, :user_url], title: I18n.t("reports.moderators_activity.labels.moderator") },
+      { properties: [:flag_count], title: I18n.t("reports.moderators_activity.labels.flag_count") },
+      { type: :seconds, properties: [:time_read], title: I18n.t("reports.moderators_activity.labels.time_read") },
+      { properties: [:topic_count], title: I18n.t("reports.moderators_activity.labels.topic_count") },
+      { properties: [:post_count], title: I18n.t("reports.moderators_activity.labels.post_count") }
+    ]
+
+    report.modes = [:table]
+
+    report.data = []
+    mod_data = {}
+
+    User.real.where(moderator: true).pluck(:id, :username).each do |u|
+      mod_data[u[0]] = {
+        user_id: u[0],
+        username: u[1],
+        user_url: "/admin/users/#{u[0]}/#{u[1]}"
       }
     end
+
+    mod_ids = mod_data.keys
+
+    return if mod_ids.empty?
+
+    time_read_query = <<~SQL
+    SELECT SUM(time_read) AS time_read,
+    user_id
+    FROM user_visits
+    WHERE user_id = ANY(ARRAY#{mod_ids})
+    AND visited_at >= '#{report.start_date}'
+    AND visited_at <= '#{report.end_date}'
+    GROUP BY user_id
+    SQL
+
+    flag_count_query = <<~SQL
+    SELECT agreed_by_id AS user_id,
+    COUNT(*) AS flag_count
+    FROM post_actions
+    WHERE (agreed_by_id = ANY(ARRAY#{mod_ids}) OR disagreed_by_id = ANY(ARRAY#{mod_ids}))
+    AND post_action_type_id = ANY(ARRAY#{PostActionType.flag_types_without_custom.values})
+    AND created_at >= '#{report.start_date}'
+    AND created_at <= '#{report.end_date}'
+    GROUP BY agreed_by_id
+    SQL
+
+    topic_count_query = <<~SQL
+    SELECT user_id,
+    COUNT(*) AS topic_count
+    FROM topics
+    WHERE user_id = ANY(ARRAY#{mod_ids})
+    AND created_at >= '#{report.start_date}'
+    AND created_at <= '#{report.end_date}'
+    GROUP BY user_id
+    SQL
+
+    post_count_query = <<~SQL
+    SELECT user_id,
+    COUNT(*) AS post_count
+    FROM posts
+    WHERE user_id = ANY(ARRAY#{mod_ids})
+    AND created_at >= '#{report.start_date}'
+    AND created_at <= '#{report.end_date}'
+    GROUP BY user_id
+    SQL
+
+    DB.query(time_read_query).each do |row|
+      mod_data[row.user_id][:time_read] = row.time_read if row.user_id
```

Comment:
`row.user_id` will never be blank since there is a `NOT NULL` constraint on the column.